core/src/main/kotlin/com/malinskiy/marathon/Marathon.kt

```diff
@@ -85,7 +86,11 @@ class Marathon(val configuration: Configuration) {
             runAsync()
         } catch (th: Throwable) {
             log.error(th.toString())
-            false
+
+            when(th) {
+                is NoDevicesException -> false
+                else -> configuration.ignoreFailures
```

Comment:
do we have any other exceptions that may occur during normal test execution? I think we should fail the build in case of any test execution error that was not caused by the tests itself. also, looks like now we will throw an exception if we have failed tests and ignoreFailures is true. maybe we should return result class here something like `(testFailures: Boolean, executionErrors: Boolean)`?

then we can throw GradleException outside if:
* `executionErrors` is true
* `cnf.ignoreFailures` is false and `testFailures` is true

If we do this, we can also customize the error message. Because now for example if we don't have devices we still print "Tests failed" error and if we haven't run any tests we can point to an empty report

Reply:
It's quite hard to distinguish between test failures and execution error in practice. NoDevices is a valid use-case that we can implement right now and I'd rather evolve as we find more exceptions than try to guess what can go wrong at this point.

I've added the `warn` message `No devices found` in case the `NoDevicesException` happens that should distinguish between no tests and no devices.

Reply:
OK, but what about the case when we have failed tests (and no exceptions during the execution) and ignoreFailures is true?
Looks like currently marathon Gradle task throws an exception in this case but it should not